### PR TITLE
Small AQL refactoring

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Small AQL refactoring, pulled out of a branch with larger changes:
- use absl::StrCat to produce error messages in `AqlCall*.cpp`.
- use newer `ExecutionEngine::execute(AqlCallStack)` API rather than old `ExecutionEngine::getSome(size_t)` API in `Query::executeV8()`. The old `getSome()` API is long superseded by the `execute()` API and should be avoided.
- remove obsolete APIs from ExecutionEngine: `getSome(size_t)`, `skipSome(size_t)`.
- modernize JavaScript code (`var` => `const` / `let`) in tests/js/server/aql/aql-optimizer-collect-aggregate.js.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: -
  - [ ] Backport for 3.10: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/AqlCall.cpp [hunk 1]
```diff
@@ -60,32 +63,31 @@ auto AqlCall::fromVelocyPack(velocypack::Slice slice) -> ResultT<AqlCall> {
 
   auto const readLimit =
       [](velocypack::Slice slice) -> ResultT<AqlCall::Limit> {
-    auto const type = slice.type();
+    auto type = slice.type();
```
Comment (Dronplane) [inline arangod/Aql/AqlCall.cpp:66]: Do we need this `type`? Why not just `slice.isString()` on the next line and remove this variable? As all way down we anyway use `slice.isXXXXX`

**Review by Dronplane — APPROVED** [review]
LGTM. 1 minor comment

File: arangod/Aql/AqlCall.cpp [hunk 2]
```diff
@@ -26,28 +26,31 @@
...
-                      slice.typeName());
+    return Result(
+        TRI_ERROR_TYPE_ERROR,
+        absl::StrCat("When deserializing AqlCall: Expected object, got ",
+                     slice.typeName()));
   }
 
-  auto expectedPropertiesFound = std::map<std::string_view, bool>{};
+  auto expectedPropertiesFound =
+      containers::FlatHashMap<std::string_view, bool>{};
```
Comment (mpoeter) [inline arangod/Aql/AqlCall.cpp:53]: Do we seriously need a FlatHashMap for this? This map only consists of _four_ fixed entries. At this size it is probably better to use an array and do a linear search. This not only avoids any and all allocations, but I know from a past investigation that the FlatHashMap produces _an awful lot of code_ for seemingly simple methods.

Comment (jsteemann) [reply arangod/Aql/AqlCall.cpp]: Done

**Review by mpoeter — APPROVED** [review]

File: arangod/Aql/AqlCall.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Aql/AqlCall.cpp]: Addressed

**Pull request merged**